warp/backend.go

```diff
@@ -15,27 +15,30 @@ import (
 	"github.com/ethereum/go-ethereum/log"
 )
 
-var _ Backend = &backend{}
-
 const batchSize = ethdb.IdealBatchSize
 
+var _ Backend = &backend{}
+
 // Backend tracks signature-eligible warp messages and provides an interface to fetch them.
 // The backend is also used to query for warp message signatures by the signature request handler.
 type Backend interface {
-	// AddMessage signs [unsignedMessage] and adds it to the warp backend database
+	// AddMessage stores [unsignedMessage] and this node's signature over it.
+	// The signature isn't persisted on disk, so if this node's BLS key changes
+	// on a subsequent startup, the signature will be recalculated and reported correctly.
 	AddMessage(unsignedMessage *avalancheWarp.UnsignedMessage) error
 
-	// GetSignature returns the signature of the requested message hash.
+	// GetSignature returns this node's signature over the message with the given [messageHash].
+	// Errors if the message wasn't added with AddMessage.
 	GetSignature(messageHash ids.ID) ([bls.SignatureLen]byte, error)
 
-	// GetMessage retrieves the [unsignedMessage] from the warp backend database if available
+	// GetMessage returns the message with the given [messageHash].
+	// Errors if the message wasn't added with AddMessage.
```

Comment:
Is the added comment necessary? What case are we trying to clarify that would be unclear to someone using this package?